src/criteria.jl

```diff
@@ -297,3 +297,29 @@ update(criterion::Patience, loss) = (loss=loss, n_increases=0)
 end
 
 done(criterion::Patience, state) = state.n_increases == criterion.n
+
+"""
+    MaximumChecks(n::Int)
+
+$STOPPING_DOC
+
+A stop is triggered by `n` consecutive checking of done.
+
+"""
+mutable struct MaximumChecks <: StoppingCriterion
+    n::Int
+    function MaximumChecks(n::Int)
+        n > 0 ||
+            throw(ArgumentError("The maxchecking level `n` must be positive. "))
+        return new(n)
+    end
+end
+MaximumChecks(; n) = MaximumChecks(n)
+
+update(criterion::MaximumChecks, loss) = 1
+update_training(criterion::MaximumChecks, loss) = 1
```

Comment:
You could drop the `update_training` overload as the fallback suffices (and this increases codecov). 

I couldn't make complete sense of the codecov report. Don't worry too much about that for now.

Reply:
Thank you, I have already remove  it.